sabnzbd/filesystem.py

```diff
@@ -166,7 +166,19 @@ def sanitize_filename(name: str) -> str:
     if not name:
         name = "unknown"
 
+    # now split name into name, ext
     name, ext = os.path.splitext(name)
+
+    # If filename is too long, truncate it:
+    if len(name) + len(ext) > 222:
+        logging.debug("Filename %s is too long, so truncating", name + ext)
+        # Too long filenames are often caused by incorrect non-ascii chars,
+        # so brute-force remove those non-ascii chars, and only keep 222 chars
+        # ... keeping in place the original extension
+        name = str(name.encode("ascii", "ignore"), "utf-8")
+        name = ''.join( c for c in name if  ord(c) >= 32 and ord(c)<128 )
```

Comment:
Unless I'm mistaken, this may significantly shorten the length of name to the point that the whole thing already fits inside the 222 limit. Even so, the code would still proceed to (unnecessarily) shorten the name by len(ext).

Reply:
I don't fully get this part. Why do we have the ascii replace and then another check using the ord()?